### DISCUSSION TITLE
Supabase Docker error - JWSError JWSInvalidSignature

### DISCUSSION THREAD

**Discussion by kavitang** [supabase/supabase#14711] [Questions]
Hey,

I'm using the Supabase Docker for local development, and everything works fine when my Next.js app accesses the DB at `http://localhost:8000`, or even when the API is accessed through Postman. 

But when I tried to run a node script (outside the app) using the Supabase client (@supabase/supabase-js package), I keep getting this error. This error is also returned by all Next `api` functions for some reason. 

I tried using both the Anon key and the Service key, I know these work because my app works fine (so does Postman). Any ideas on what could be causing this and how this can be resolved?

```
 {
  code: 'PGRST301',
  details: null,
  hint: null,
  message: 'JWSError JWSInvalidSignature'
}
```

EDIT: The same Node script works if I switch to `fetch` and use the full API path `http://localhost:8000/rest/v1/table_name?select=id`, with the ApiKey header using the same Service key. The issue seems to be specifically the Supabase client created by supabase-js.

EDIT 2: As a temporary fix I'm working with `@supabase/postgrest-js` package directly, which is working absolutely fine.

Thanks!

**Comment by infinito84** [comment]
In my case the error was due I was using a JWT_SECRET too short, using the generated from here https://supabase.com/docs/guides/self-hosting#api-keys works very  well

**Reply by kavitang** [reply]
Thanks @infinito84 but the requests work completely fine with the `@supabase/postgrest-js` package as well as the `@supabase/supabase-js` package (when used in the client app). It seems unlikely the JWT_SECRET would work in one case and not another..
